Composer/packages/client/src/pages/publish/BotStatusList.tsx

```diff
@@ -7,82 +7,55 @@ import moment from 'moment';
 import formatMessage from 'format-message';
 import { Checkbox } from 'office-ui-fabric-react/lib/Checkbox';
 import { Icon } from 'office-ui-fabric-react/lib/Icon';
-import React, { useState, Fragment } from 'react';
+import React, { useState, Fragment, useMemo } from 'react';
 import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
 import { Spinner, SpinnerSize } from 'office-ui-fabric-react/lib/Spinner';
-import { PublishTarget, PublishResult } from '@bfc/shared';
-import { CheckboxVisibility, DetailsList, IColumn } from 'office-ui-fabric-react/lib/DetailsList';
+import { PublishResult } from '@bfc/shared';
+import { CheckboxVisibility, DetailsList } from 'office-ui-fabric-react/lib/DetailsList';
 import { IconButton } from 'office-ui-fabric-react/lib/Button';
 import { SharedColors } from '@uifabric/fluent-theme';
 import { FontSizes } from '@uifabric/styling';
+import get from 'lodash/get';
 
-import { navigateTo } from '../../utils/navigation';
-import { PublishType } from '../../recoilModel/types';
+import { ApiStatus } from '../../utils/publishStatusPollingUpdater';
 
 import { PublishStatusList } from './PublishStatusList';
 import { detailList, listRoot, tableView } from './styles';
+import { BotPublishHistory, BotStatus } from './type';
 
-export type IBotStatus = {
-  id: string;
-  name: string;
-  publishTargets?: PublishTarget[];
-  publishTarget?: string;
-  time?: string;
-  status?: number;
-  message?: string;
-  comment?: string;
-};
-export type IBotStatusListProps = {
-  projectId: string;
-  items: IBotStatus[];
-  botPublishHistoryList: { projectId: string; publishHistory: { [key: string]: PublishResult[] } }[];
-  botPublishTypesList: { projectId: string; publishTypes: PublishType[] }[];
-  publishDisabled: boolean;
-  updateItems: (items: IBotStatus[]) => void;
-  updatePublishHistory: (items: PublishResult[], item: IBotStatus) => void;
-  updateSelectedBots: (items: IBotStatus[]) => void;
-  changePublishTarget: (PublishTarget: string, item: IBotStatus) => void;
-  onLogClick: (item: PublishResult) => void;
-  onRollbackClick: (selectedVersion: PublishResult, item: IBotStatus) => void;
-};
+export interface BotStatusListProps {
```

Comment:
use type instead of interface

Reply:
In most of the practices I can find, props of a JSX is declared as an interface. Do we have any reference of interface vs type?

Reply:
One of the reason I prefer to use `interface` over `type` is it can trigger VSCode's intellisense:
![image](https://user-images.githubusercontent.com/8528761/104691403-20388d80-5741-11eb-9d96-5b09a1aae05a.png)